### PR TITLE
[ci] Retry all failed test steps if 'retry-flaky-tests' CI tag is set.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: scripts/ci/run-step.sh [hunk 1]
```diff
@@ -26,6 +26,10 @@ case $i in
     TIMEOUT="${i#*=}"
     shift # past argument=value
     ;;
+    --retry=*)
+    RETRY="${i#*=}"
+    shift
+	;;
```
Comment (akoeplinger) [inline scripts/ci/run-step.sh]: we apparently use spaces for indentation in this file, please keep it consistent.

File: scripts/ci/run-step.sh [hunk 2]
```diff
@@ -52,9 +56,29 @@ if [ -z "${LABEL}" ] || [ -z "${TIMEOUT}" ]
...
+if [ -n "${RETRY}" ]; then
+	COUNT=${RETRY}
 fi
+
+ITER=0
+while [ ${ITER} -lt ${COUNT} ]; do
+	if [ ${ITER} -eq 0 ]; then
+		ITER_LABEL=${LABEL}
+	else
+		ITER_LABEL="${LABEL} (retry)"
```
Comment (akoeplinger) [inline scripts/ci/run-step.sh]:
```suggestion
		ITER_LABEL="${LABEL} (retry ${ITER})"
```

**Comment by vargaz** [comment]
There are multiple ways of solving this, this seemed like the easiest. One test step usually doesn't take very long, its the restarting of the whole lane which takes a long time.

**Comment by vargaz** [comment]
So the retry seems to work:
https://jenkins.mono-project.com/job/test-mono-pull-request-armhf/27258/
But the lane is still marked as failed.

**Comment by vargaz** [comment]
Old.

**Pull request closed without merging**